src/Controller/Admin/ProductController.php

```diff
@@ -0,0 +1,158 @@
+<?php
+
+namespace App\Controller\Admin;
+
+use App\Entity\Product;
+use App\Form\ProductType;
+use App\Repository\ProductRepository;
+use App\Repository\CategoryRepository;
+use App\Repository\OrderProductRepository;
+use App\Repository\ProductFeedbackRepository;
+use Doctrine\ORM\EntityManagerInterface;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Annotation\Route;
+use Symfony\Component\Filesystem\Filesystem;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\IsGranted;
+use App\Manager\CartManager;
+
+/**
+ * @Route("/product")
+ */
+class ProductController extends AbstractController
+{
+    /**
+     * @Route("/", name="admin_product_index", methods={"GET"}, host="admin.%domain%")
+     * @IsGranted("ROLE_ADMIN")
+     */
+    public function index(ProductRepository $productRepository): Response
+    {
+        return $this->render('admin/product/index.html.twig', [
+            'products' => $productRepository->findAll(),
+        ]);
+    }
+
+    /**
+     * @Route("/new", name="admin_product_new", methods={"GET","POST"}, host="admin.%domain%")
+     */
+    public function new(Request $request, EntityManagerInterface $entityManager, CategoryRepository $categoryRepository): Response
+    {
+        $product = new Product();
+        $form = $this->createForm(ProductType::class, $product);
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            $product->setUser($this->getUser());
+            $product->setCategory($form->get('cat')->getData());
+
+            $file = $form->get('image')->getData();
+
+            if ($file) {
+                $originalFilename = pathinfo($file->getClientOriginalName(), PATHINFO_FILENAME);
+                // this is needed to safely include the file name as part of the URL
+                $newFilename = md5(uniqid()).'.'.$file->guessExtension();
+
+                // Move the file to the directory where brochures are stored
+                try {
+                    $file->move(
+                        $this->getParameter('product_images_directory'),
+                        $newFilename
+                    );
+
+                    $product->setImageLink($newFilename);
+                } catch (FileException $e) {
+                    // ... handle exception if something happens during file upload
```

Comment:
you missed to add logic here ? please update catch condition.

Reply:
done